### DISCUSSION TITLE
TVP as Constraints

### DISCUSSION THREAD

**Discussion by dsdhms** [do-mpc/do-mpc#225] [Q&A]
Is it possible to use Time-Varying Parameters (TVP) as right-hand-side values for lower and/or upper bounds of constraints?

Regards,
dsdhms

**Comment by 4flixt** [comment]
Hi @dsdhms ,

yes, you can use tvp values as constraints but not by setting them as upper/lower values for the bounds on optimization variables. These bounds are static.
Instead you must work with ``mpc.set_nl_cons`` which takes an arbitrary expression and an upper bound. Use your ``tvp`` values in the expression and a fixed bound to yield time-varying constraints.

Best
4flixt
